federation/pkg/federation-controller/sync/controller.go

```diff
@@ -452,22 +486,23 @@ func syncToClusters(clustersAccessor clustersAccessorFunc, operationsAccessor op
 		return statusError
 	}
 
-	// Evertyhing is in order but let's be double sure
+	// Everything is in order but let's be double sure
 	return statusNeedsRecheck
 }
 
 type clusterObjectAccessorFunc func(clusterName string) (interface{}, bool, error)
 type clusterSelectorFunc func(map[string]string, map[string]string) (bool, error)
 
 // clusterOperations returns the list of operations needed to synchronize the state of the given object to the provided clusters
-func clusterOperations(adapter federatedtypes.FederatedTypeAdapter, clusters []*federationapi.Cluster, obj pkgruntime.Object, key string, accessor clusterObjectAccessorFunc, selector clusterSelectorFunc) ([]util.FederatedOperation, error) {
-	// The data should not be modified.
-	desiredObj := adapter.Copy(obj)
-	objMeta := adapter.ObjectMeta(desiredObj)
+func clusterOperations(adapter federatedtypes.FederatedTypeAdapter, clusters []*federationapi.Cluster, obj pkgruntime.Object, key string, userInfo interface{}, accessor clusterObjectAccessorFunc, selector clusterSelectorFunc) ([]util.FederatedOperation, error) {
 	kind := adapter.Kind()
 	operations := make([]util.FederatedOperation, 0)
 
 	for _, cluster := range clusters {
+		// The data should not be modified.
+		desiredObj := adapter.Copy(obj)
+		objMeta := adapter.ObjectMeta(desiredObj)
+
 		clusterObj, found, err := accessor(cluster.Name)
```

Comment:
Hmm. I think I see your point. One potential way we could address this is by getting the clusters once in a loop at the beginning of reconcile() and using that list of clusters throughout the method. Would that solve your issue? The entire reconcile loop would use the same cluster objects in that case.

I think your point about having a minimal interface is valid, but I don't think that the smallest number of methods or amount of data passed around is necessarily the most minimal interface. Having one method in an interface which has semantic complexity and is hard to understand is less "minimal" than having more methods that are more like a unit.

Also, I think I forgot to repush this branch: I changed these methods quite a bit to be more scheduling specific. I'll do that now.